src/PackageManagement/IDE/PackageRestoreManager.cs

```diff
@@ -25,13 +25,18 @@ public class PackageRestoreManager : IPackageRestoreManager
 
         private ISourceRepositoryProvider SourceRepositoryProvider { get; }
         private ISolutionManager SolutionManager { get; }
+        private IDeleteOnRestartManager DeleteOnRestartManager { get; }
         private ISettings Settings { get; }
 
         public event EventHandler<PackagesMissingStatusEventArgs> PackagesMissingStatusChanged;
         public event EventHandler<PackageRestoredEventArgs> PackageRestoredEvent;
         public event EventHandler<PackageRestoreFailedEventArgs> PackageRestoreFailedEvent;
 
-        public PackageRestoreManager(ISourceRepositoryProvider sourceRepositoryProvider, ISettings settings, ISolutionManager solutionManager)
+        public PackageRestoreManager(
+            ISourceRepositoryProvider sourceRepositoryProvider,
+            ISettings settings,
+            ISolutionManager solutionManager,
+            IDeleteOnRestartManager deleteOnRestartManager)
```

Comment:
lets clean this up when we remove the reference to the scenarios where solution manager is not present.